packages/salesforcedx-vscode-core/src/orgPicker/orgList.ts

```diff
@@ -88,34 +90,26 @@ export class OrgList implements vscode.Disposable {
         // scratch orgs parented by other (non-default) devHub orgs
         continue;
       }
-      const isExpired =
-        authFields && authFields.expirationDate
-          ? today >= new Date(authFields.expirationDate)
-          : false;
-
-      const aliases = await ConfigUtil.getAllAliasesFor(orgAuth.username);
-      let authListItem =
-        aliases && aliases.length > 0
-          ? `${aliases.join(',')} - ${orgAuth.username}`
-          : orgAuth.username;
 
-      if (isExpired) {
-        authListItem += ` - ${nls.localize(
-          'org_expired'
-        )} ${String.fromCodePoint(0x274c)}`; // cross-mark
+      if (orgAuth.isExpired === true) {
+        // If the scratch org is expired we don't want to see it in the org picker
+        continue;
       }
+      const aliases = orgAuth.aliases || [];
+      const authListItem =
+        aliases?.length > 0
+          ? `${aliases.join(',')} - ${orgAuth.username}`
+          : orgAuth.username;
 
       authList.push(authListItem);
     }
     return authList;
   }
 
   public async updateOrgList(): Promise<string[]> {
-    const orgAuthorizations = await this.getOrgAuthorizations();
-    if (orgAuthorizations && orgAuthorizations.length === 0) {
-      return [];
-    }
-    const authUsernameList = await this.filterAuthInfo(orgAuthorizations);
+    const authUsernameList = await this.filterAuthInfo(
```

Comment:
You skipped the check if `orgAuthorizations` is empty here, which will always have a call of `filterAuthInfo`. In `filterAutoInfo`, there will always be a `const targetDevHub = await OrgAuthInfo.getDevHubUsername();` which might be dummy.

Reply:
Because if orgAuthorizations is empty filterAuthInfo will return an empty array anyways. See https://github.com/forcedotcom/salesforcedx-vscode/pull/5729#discussion_r1727107139